### ISSUE TITLE
Different graph from example

### ISSUE THREAD

**Issue by RemDelaporteMathurin** [JosephBARBIERDARNAL/pypalettes#29]
Hi @JosephBARBIERDARNAL thanks for this amazing tool. I keep recommending it to everyone.

Wanted to make a treemap plot today and tried running the example code given on the website:

```python
# load libraries
import squarify # pip install squarify (algorithm for treemap)
import matplotlib.pyplot as plt
from pypalettes import load_cmap
import pandas as pd

df = pd.read_csv("https://raw.githubusercontent.com/holtzy/The-Python-Graph-Gallery/master/static/data/simple-treemap.csv")

# create a color palette
cmap = load_cmap('Chevalier1')
category_codes, unique_categories = pd.factorize(df['parent'])
colors = [cmap(code / (len(unique_categories) - 1)) for code in category_codes]

# create a treemap
fig, ax = plt.subplots(figsize=(11,8), dpi=300)
ax.set_axis_off()
squarify.plot(
   sizes=df["value"],
   label=df["name"],
   color=colors,
   text_kwargs={'color':'white'},
   pad=True,
   ax=ax
)
plt.show()
```

I was expecting this:
<img width="590" alt="image" src="https://github.com/user-attachments/assets/8809d827-8457-4110-a378-54ba783c166d">

But got:
![image](https://github.com/user-attachments/assets/67c2e14e-c1f1-4078-a70d-cd910d4910a1)

Is it something you can reproduce on your end? I should be using (haven't checked) latest versions of everything

**Comment by RemDelaporteMathurin** [comment]
After talking with @JosephBARBIERDARNAL it sounds like this may be an issue in `squarify` maybe we should move the issue there? https://github.com/laserson/squarify

Ping @laserson

**Comment by laserson** [comment]
On the surface, it looks like things maybe are out of order? Are you giving a sorted list to squarify? Did the categories/colors somehow get permuted?

**Comment by RemDelaporteMathurin** [comment]
This is what the dataframe looks like:

<div>

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>type</th>
      <th>name</th>
      <th>value</th>
      <th>parent</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>leaf</td>
      <td>Mark</td>
      <td>90</td>
      <td>Team Dataviz</td>
    </tr>
    <tr>
      <th>1</th>
      <td>leaf</td>
      <td>Robert</td>
      <td>12</td>
      <td>Team Dataviz</td>
    </tr>
    <tr>
      <th>2</th>
      <td>leaf</td>
      <td>Emily</td>
      <td>34</td>
      <td>Team Dataviz</td>
    </tr>
    <tr>
      <th>3</th>
      <td>leaf</td>
      <td>Marion</td>
      <td>53</td>
      <td>Team Dataviz</td>
    </tr>
    <tr>
      <th>4</th>
      <td>leaf</td>
      <td>Nicolas</td>
      <td>98</td>
      <td>Team DevOps</td>
    </tr>
    <tr>
      <th>5</th>
      <td>leaf</td>
      <td>Malki</td>
      <td>22</td>
      <td>Team DevOps</td>
    </tr>
    <tr>
      <th>6</th>
      <td>leaf</td>
      <td>Djé</td>
      <td>12</td>
      <td>Team DevOps</td>
    </tr>
    <tr>
      <th>7</th>
      <td>leaf</td>
      <td>Mélanie</td>
      <td>45</td>
      <td>Team Sales</td>
    </tr>
    <tr>
      <th>8</th>
      <td>leaf</td>
      <td>Einstein</td>
      <td>76</td>
      <td>Team Sales</td>
    </tr>
  </tbody>
</table>
</div>


If it is permutated at some point it's not before the `squarify.plot` call

**Comment by laserson** [comment]
Check out this PR:
https://github.com/laserson/squarify/pull/31

**Comment by RemDelaporteMathurin** [comment]
> Looks pretty clear to me that the treemap viz on the website is not generated by the example code. The viz on the website looks like it's been treemapped hierarchically, first by grouping on "parent", and then internally to each group.

Right this is what we figured out it was generated with D3.js

Thanks for the pointers!

**Comment by RemDelaporteMathurin** [comment]
Ah would be great is this was a functionality of `squarify`. It would take me some tinkering to generate plots like this: 
![image](https://github.com/user-attachments/assets/78f63d51-c238-426b-b626-967d74956042)

**Comment by laserson** [comment]
Indeed that was the subject matter of the (unmerged) PR I linked to, but it started making things a bit too complicated. That said, it shouldn't be too hard to implement it specifically for your own use case.

**Comment by RemDelaporteMathurin** [comment]
By the looks of it the PR added a new function plot_subgroup that I have to tinker with and understand as well as modifications to other functions.

It would take me some time (that I don't have atm unfortunately) to get that to work even it isn't too hard 😄

**Comment by JosephBARBIERDARNAL** [comment]
Thanks for your input here @laserson ! Yes, it's our fault (me and @holtzy) that the treemap image doesn't match what squarify actually does.

Just a naive question about the feature from the PR you mentioned, why are you trying to avoid adding complexity? I have no idea how squarify works under the bonnet, but do you think it would be too difficult to maintain? Or you want to keep focus on a single feature? Intuitively it seems like a good addition to your package.

**Comment by laserson** [comment]
As far as I'm concerned the only real functionality of the squarify package is the `squarify` function. The `plot` function was added as a quick-n-dirty convenience for people, but I'm not really interested in maintaining a more feature-filled version of that.

**Comment by RemDelaporteMathurin** [comment]
@JosephBARBIERDARNAL maybe there is another package to be made here! 👀 Could add it to https://python-graph-gallery.com/

**Comment by JosephBARBIERDARNAL** [comment]
@RemDelaporteMathurin yes haha! There are so many packages I'd love to spend time on, but unfortunately I only have 24 hours a day

**Comment by JosephBARBIERDARNAL** [comment]
Closing this!
